code/controllers/subsystem/dynamic/_dynamic_ruleset.dm

```diff
@@ -0,0 +1,361 @@
+ */
+/datum/dynamic_ruleset
+	/// Human-readable name of the ruleset.
+	var/name
+	/// Tag the ruleset uses for configuring.
+	/// Don't change this unless you know what you're doing.
+	var/config_tag
+
+	/// What flag to check for jobbans? Optional, if unset, uses pref_flag
+	var/jobban_flag
+	/// What flag to check for prefs? Required if the antag has an associated preference
+	var/pref_flag
+	/// Flags for this ruleset
+	var/ruleset_flags = NONE
+	/// Points to what antag datum this ruleset will use for generating a preview icon in the prefs menu
+	var/preview_antag_datum
+	/// List of all minds selected for this ruleset
+	VAR_FINAL/list/datum/mind/selected_minds = list()
+
+	/**
+	 * The chance the ruleset is picked when selecting from the pool of rulesets.
+	 *
+	 * This can either be
+	 * - A list of weight corresponding to dynamic tiers.
+	 * If a tier is not specified, it will use the next highest tier.
+	 * Or
+	 * - A single weight for all tiers.
+	 */
+	var/list/weight = 0
+	/**
+	 * The min population for which this ruleset is available.
+	 *
+	 * This can either be
+	 * - A list of min populations corresponding to dynamic tiers.
+	 * If a tier is not specified, it will use the next highest tier.
+	 * Or
+	 * - A single min population for all tiers.
+	 */
+	var/list/min_pop = 0
+	/// List of roles that are blacklisted from this ruleset
+	/// For roundstart rulesets, it will prevent players from being selected for this ruleset if they have one of these roles
+	/// For latejoin or midround rulesets, it will prevent players from being assigned to this ruleset if they have one of these roles
+	var/list/blacklisted_roles = list()
+	/**
+	 * How many candidates are needed for this ruleset to be selected?
+	 * Ie. "We won't even bother attempting to run this ruleset unless at least x players want to be it"
+	 *
+	 * This can either be
+	 * - A number
+	 * Or
+	 * - A list in the form of list("denominator" = x, "offset" = y)
+	 * which will divide the population size by x and add y to it to calculate the number of candidates
+	 */
+	var/min_antag_cap = 1
+	/**
+	 * How many candidates will be this ruleset try to select?
+	 * Ie. "We have 10 cadidates, but we only want x of them to be antags"
+	 *
+	 * This can either be
+	 * - A number
+	 * Or
+	 * - A list in the form of list("denominator" = x, "offset" = y)
+	 * which will divide the population size by x and add y to it to calculate the number of candidates
+	 *
+	 * If null, defaults to min_antag_cap
+	 */
+	var/max_antag_cap
+	/// If set to TRUE, dynamic will be able to draft this ruleset again later on
+	var/repeatable = FALSE
+	/// Every time this ruleset is selected, the weight will be decreased by this amount
+	var/repeatable_weight_decrease = 2
+	/// Players whose account is less than this many days old will be filtered out of the candidate list
+	var/minimum_required_age = 0
+	/// Templates necessary for this ruleset to be executed
+	var/list/ruleset_lazy_templates
```

Comment:
Maybe stuff like this should be in VAR_PROTECTED to signal that they shouldn't be edited by config.